lib/engine-object-file/src/artifact.rs

```diff
@@ -319,10 +319,12 @@ impl ObjectFileArtifact {
 
             byte_buffer[0..WORD_SIZE]
                 .clone_from_slice(&bytes[cur_offset..(cur_offset + WORD_SIZE)]);
+            let fp = usize::from_ne_bytes(byte_buffer);
```

Comment:
Can you explain what is going on here?

Reply:
`bytes` is a slice of bytes, at some offset (`cur_offset`) into that stream of bytes, there is a pointer. We grab the pointer out of that stream of bytes and ultimately cast it to FunctionBodyPtr.

I don't know why we're doing this, but this is what the code was already doing. I could look into why if you want?

Reply:
This code is parsing in metadata stored in the object file, it's creating types like `usize` from the buffer of bytes